kubernetes-deployment/massdriver.yaml

```diff
@@ -46,50 +67,62 @@ params:
           type: integer
           default: 90
     ingress:
+      title: Ingress
+      description: Configure the application for public
       type: object
       properties:
         enabled:
+          title: Enable Public Internet Access
           type: boolean
         host:
+          title: Hostname
+          description: TBD, is this the right field name?
           type: string
         path:
           type: string
           default: '/'
-    envs:
-      type: array
-      items:
-        type: object
-        properties:
-          name:
-            type: string
-          value:
-            type: string
-<md if not .Connections md>
-connections:
-  properties: {}
-<md end -md>
 
-<md if .Connections md>
+# TODO: if this fails during template rendering because .Conditions isn't present, we shouldn't propogate nils, it should return an empty map
 connections:
   required:
+    - kubernetes_cluster
   <md- range $key, $art:= .Connections md>
     - <md $key md>
   <md- end md>
+  oneOf:
```

Comment:
Does `oneOf` work here on connections? I vaguely remember I tried this exact thing back when we were first trying out applications and it crashed the UI when I dragged it on the canvas.

Reply:
We aren’t saying oneOf a connection, we are saying oneOf a requirement.